server/project.go

```diff
@@ -244,10 +244,6 @@ func (s *Server) registerProjectRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusBadRequest, "Malformed create linked repository request").SetInternal(err)
 		}
 
-		if strings.Contains(repositoryCreate.BranchFilter, "*") {
```

Comment:
Branch filter is only supported in GitLab, so this check still applies to GitHub

Reply:
We moved the branch filtering from GitLab/GitHub to Bytebase, so the behavior remains consistent.